### DISCUSSION TITLE
Question: Weights Decreasing using PostPre Learning Rule

### DISCUSSION THREAD

**Discussion by tasha061** [BindsNET/bindsnet#603] [General]
Hi,
I am using the PostPre learning rule in BindsNet and I am getting good accuracy from my SNN network. However, my weights seem to be decreasing instead of increasing is this normal or would you know a reason why this would be happening?

Many thanks.

**Comment by SimonInParis** [comment]
Is the average of the weights decreasing? or only some of them?
Did you put a constraint on the norm of the layer's weights?
Keep in mind that increase/decrease of the w is also related to the amount of incoming/outgoing spikes. You can check that too.
